drivers: adc: add LPADC driver support to lpc55s36

dleach02 opened the pull request:
Add LPADC support to the lpcxpresso55s36 platform.

decsny commented on soc/arm/nxp_lpc/lpc55xxx/Kconfig.soc:
```diff
@@ -53,6 +53,7 @@ config SOC_LPC55S36
 	select ARMV8_M_DSP
 	select ARM_TRUSTZONE_M
 	select CLOCK_CONTROL
+	select HAS_MCUX_LPADC
```
This is unnecessary to select here, it will be enabled by default because of [devicetree kconfig macros](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/adc/Kconfig.mcux), better to have less selects across the tree to have less points to maintain

decsny commented on samples/drivers/adc/boards/lpcxpresso55s36.overlay:
```diff
@@ -0,0 +1,33 @@
+		zephyr,acquisition-time = <ADC_ACQ_TIME_DEFAULT>;
+		zephyr,vref-mv = <3300>;
+		zephyr,resolution = <12>;
+	};
```
there needs to be a `zephyr,input-positive` property here to meet the requirements of the new [LPADC driver rewrite](https://github.com/zephyrproject-rtos/zephyr/commit/e587047dc6b77c28ea399e272c41beb2cd52f142)

decsny commented on samples/drivers/adc/boards/lpcxpresso55s36.overlay:
```diff
@@ -0,0 +1,33 @@
+ */
+
+#include <zephyr/dt-bindings/adc/adc.h>
+
```
need `#include <zephyr/dt-bindings/adc/mcux-lpadc.h>`

decsny commented on samples/drivers/adc/boards/lpcxpresso55s36.overlay:
```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2022 NXP
```
2023*

decsny commented on dts/arm/nxp/nxp_lpc55S3x_common.dtsi:
```diff
@@ -227,6 +227,21 @@
+		offset-value-b = <10>;
+		#io-channel-cells = <1>;
+	};
+
```
there is also a second LPADC called `adc1` on this chip, can we add that here

BerkantCiftci replied:
Hi @decsny , I was looking into trying to get ADC1 working since I want to make use of it. I was wondering if this is still something planned and if it would help if I can share the part I think that needs to be in there in a seperate PR. It took me some time and Im not sure if I did it right.

decsny commented on drivers/adc/adc_mcux_lpadc.c:
```diff
@@ -423,6 +423,10 @@ static int mcux_lpadc_init(const struct device *dev)
 	CLOCK_AttachClk(kFRO_DIV4_to_ADC_CLK);
 	CLOCK_SetClkDiv(kCLOCK_DivAdcClk, 1);
+
+#elif defined(CONFIG_SOC_LPC55S36)
```
we can do this in the soc.c instead to avoid adding to the technical debt related to the clocking in this file

dleach02 replied:
Lets do a separate PR to fix this completely... we will still need to have a change here to keep the #if/#else to work correctly for the 55s36 if we did the clock stuff in the soc.c

decsny reviewed (dismissed):
resolving testing issues

Commit:
drivers: lpadc: Enable VREF in init

Enable VREF peripheral in the LPADC init
so that ADC can measure voltages properly.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
drivers: adc: Add LPADC driver support to lpc55s36

Add LPADC support to LPC55S36 SOC platform

Signed-off-by: David Leach <david.leach@nxp.com>

Commit:
boards: lpcxpresso55s36: Enable LPADC

Enable LPADC on lpcxpresso55s36 board
Support lpcxpresso55s36 on ADC driver sample

Signed-off-by: David Leach <david.leach@nxp.com>

decsny commented:
> @decsny I'd like to see this squashed into one commit- as the PR stands right now, the LPADC is enabled in the first commit, and then the second commit implements vref support. This means that the ADC is initially enabled (and broken for one commit), then fixed in the prior commit. I'd prefer one commit so that if a user is bisecting in the future, the commit adding the ADC support also functions correctly (rather than the next commit being needed)
> 
> Maybe you can use [co-authoring](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors) to indicate both you and David worked on this enablement?

Is  the new commit history alright for you?
